In [1]:
!pip install pulp

In [2]:
from pulp import LpMaximize, LpProblem, LpVariable, lpSum, LpStatus

In [3]:
model = LpProblem(name="factory-production", sense=LpMaximize)

In [4]:
# x = Product A units, y = Product B units
x = LpVariable(name="Product_A", lowBound=0, cat='Integer')
y = LpVariable(name="Product_B", lowBound=0, cat='Integer')

In [5]:
# Constraint 1: Machine time
model += (2 * x + 4 * y <= 100, "Machine_Time")

# Constraint 2: Raw materials
model += (3 * x + 2 * y <= 90, "Raw_Material")

In [6]:
# Objective: Maximize profit = 20x + 30y
model += 20 * x + 30 * y


In [7]:
status = model.solve()

In [8]:
print(f"Status: {LpStatus[model.status]}")
print(f"Produce {x.value()} units of Product A")
print(f"Produce {y.value()} units of Product B")
print(f"Maximum Profit: ₹{model.objective.value()}")

Status: Optimal
Produce 20.0 units of Product A
Produce 15.0 units of Product B
Maximum Profit: ₹850.0


### 📊 Business Insights

- The optimal strategy is to produce **20 units of Product A** and **15 units of Product B**.
- This production plan will **maximize the profit to ₹850** while satisfying all resource constraints (labor and material).
- The optimization model helps the business **utilize resources efficiently** and **make data-driven production decisions**.
